lib/patches/rake-compiler/rake/extensiontask.rb

```diff
@@ -0,0 +1,18 @@
+Truffle::Patching.require_original __FILE__
+
+module Rake
+  class ExtensionTask < BaseExtensionTask
+    def init(name = nil, gem_spec = nil)
+      super
+      @config_script = 'extconf.rb'
+      @source_pattern = "*.{c,cc,cpp}"
+      @compiled_pattern = "*.{bc,su}"
+      @cross_compile = false
+      @cross_config_options = []
+      @cross_compiling = nil
+      @no_native = false
+      @config_includes = []
+      @ruby_versions_per_platform = {}
+    end
```

Comment:
I think this would be much cleaner with a prepended module, so then we can call super and just re-write the `@compiled_pattern`, like in https://github.com/graalvm/truffleruby/blob/8966486bea4c776956fd388a8edcbca389557be5/lib/patches/stdlib/rubygems/request.rb
In general, I think it's good for all patches to adopt this style, because it avoids copying too much code and it's also clearer where the method comes from when using `#method`.

Reply:
I've updated this to use the prepended module, thanks.